File: static/js/suppliers.js
// Gestion des fournisseurs - Version 2.3 - Fixed empty state display
console.log('Suppliers.js loaded - Version 2.3');

let suppliers = [];
let currentSupplierId = null;
let isEditMode = false;

// Initialisation
document.addEventListener('DOMContentLoaded', function () {
    console.log('DOM loaded, checking elements...');

    // Vérifier et créer les éléments manquants si nécessaire
    ensureRequiredElements();

    console.log('After ensuring elements:');
    console.log('suppliersContainer:', document.getElementById('suppliersContainer'));
    console.log('emptyState:', document.getElementById('emptyState'));
    console.log('suppliersList:', document.getElementById('suppliersList'));

    // Même logique que products.js pour attendre l'auth prête (cookies HttpOnly)
    const ready = () => {
        const hasAuthManager = !!window.authManager;
        const hasUser = !!(hasAuthManager && window.authManager.userData && Object.keys(window.authManager.userData).length);
        return hasAuthManager && (window.authManager.isAuthenticatedSync() || hasUser);
    };

    // Initialiser immédiatement sans délai pour un chargement instantané
    loadSuppliers();
    setupEventListeners();
});

// S'assurer que les éléments requis existent
function ensureRequiredElements() {
    let container = document.getElementById('suppliersContainer');

    // Si le container principal n'existe pas, le créer
    if (!container) {
        console.log('Creating suppliersContainer...');
        const contentDiv = document.querySelector('.container-fluid') || document.querySelector('main') || document.body;
        container = document.createElement('div');
        container.id = 'suppliersContainer';
        container.className = 'mt-4';
        contentDiv.appendChild(container);
    }

    // S'assurer que emptyState existe
    let emptyState = document.getElementById('emptyState');
    if (!emptyState) {
        console.log('Creating emptyState...');
        emptyState = document.createElement('div');
        emptyState.id = 'emptyState';
        emptyState.className = 'text-center py-5';
        emptyState.innerHTML = `
            <i class="bi bi-truck display-1 text-muted"></i>
            <h4 class="text-muted mt-3">Aucun fournisseur</h4>
            <p class="text-muted">Commencez par ajouter votre premier fournisseur</p>
        `;
        container.appendChild(emptyState);
    }

    // S'assurer que suppliersList existe
    let suppliersList = document.getElementById('suppliersList');
    if (!suppliersList) {
        console.log('Creating suppliersList...');
        suppliersList = document.createElement('div');
        suppliersList.id = 'suppliersList';
        suppliersList.className = 'row';
        suppliersList.style.display = 'none';
        container.appendChild(suppliersList);
    }

    // S'assurer que resultsCount existe
    let resultsCount = document.getElementById('resultsCount');
    if (!resultsCount) {
        console.log('Creating resultsCount...');
        const searchSection = document.querySelector('.row.mb-4');
        if (searchSection) {
            const countDiv = document.createElement('div');
            countDiv.className = 'col-md-6 text-end';
            countDiv.innerHTML = '<small class="text-muted" id="resultsCount">0 fournisseur trouvé</small>';
            searchSection.appendChild(countDiv);
        }
    }
}

// Configuration des écouteurs d'événements
function setupEventListeners() {
    // Recherche en temps réel
    const searchInput = document.getElementById('searchInput');
    if (searchInput) {
        searchInput.addEventListener('input', debounce(filterSuppliers, 300));
    }

    // Formulaire de soumission
    const form = document.getElementById('supplierFormElement');
    if (form) {
        form.addEventListener('submit', handleFormSubmit);
    }

    // Modal de suppression
    const confirmDeleteBtn = document.getElementById('confirmDeleteBtn');
    if (confirmDeleteBtn) {
        confirmDeleteBtn.addEventListener('click', confirmDelete);
    }
}

/*
  Pagination de l'affichage.

  L'écran déversait tous les fournisseurs d'un bloc — sur une centaine de
  fiches, un mur de cartes. Les tuiles de statistiques doivent en revanche
  compter l'ensemble, pas la page affichée : on charge donc tout en une fois
  (un fournisseur est une petite table de référence) et on découpe seulement le
  rendu. Le total renvoyé en en-tête sert à détecter le cas où le plafond de
  l'API serait atteint.
*/
let suppliersPage = 1;
const SUPPLIERS_PER_PAGE = 12;
const SUPPLIERS_FETCH_LIMIT = 1000;

// Charger les fournisseurs
async function loadSuppliers() {
    try {
        showLoading();

        // Appel API via utilitaire avec timeout pour éviter les chargements infinis
        const response = await safeLoadData(
            () => axios.get('/api/suppliers/', { params: { limit: SUPPLIERS_FETCH_LIMIT } }),
            {
                timeout: 8000,
                fallbackData: [],
                errorMessage: 'Erreur lors du chargement des fournisseurs'
            }
        );

        const payload = response?.data ?? {};
        suppliers = Array.isArray(payload) ? payload : (payload.suppliers || []);

        // `X-Total-Count` : le corps reste une liste typée, le total voyage à côté.
        const total = Number(response?.headers?.['x-total-count']) || suppliers.length;
        if (total > suppliers.length) {
            // Mieux vaut le dire que d'afficher des statistiques fausses en silence.
            showError(`${suppliers.length} fournisseurs affichés sur ${total} : affinez la recherche.`);
        }

        suppliersPage = 1;
        hideLoading();
        displaySuppliers();
        updateStatistics();
    } catch (error) {
        console.error('Erreur lors du chargement des fournisseurs:', error);
        hideLoading();
        showError(error.response?.data?.detail || 'Erreur lors du chargement des fournisseurs');
        hideLoading();
    }
}

// Afficher les fournisseurs
function displaySuppliers(suppliersToShow) {
    console.log('displaySuppliers called with:', suppliersToShow);
    console.log('Type of suppliersToShow:', typeof suppliersToShow);
    console.log('Is array:', Array.isArray(suppliersToShow));

    // S'assurer que les éléments existent avant de les utiliser
    ensureRequiredElements();

    const container = document.getElementById('suppliersContainer');
    const emptyState = document.getElementById('emptyState');
    const suppliersList = document.getElementById('suppliersList');

    console.log('Elements found:', {
        container: !!container,
        emptyState: !!emptyState,
        suppliersList: !!suppliersList
    });

    // Les éléments devraient maintenant exister grâce à ensureRequiredElements
    if (!container || !emptyState || !suppliersList) {
        console.error('Critical error: Required elements still missing after ensureRequiredElements');
        return;
    }

    // Par défaut, utiliser le tableau global si aucun paramètre n'est fourni
    if (!Array.isArray(suppliersToShow)) {
        suppliersToShow = Array.isArray(suppliers) ? suppliers : [];
    }

    if (suppliersToShow.length === 0) {
        emptyState.style.display = 'block';
        suppliersList.style.display = 'none';
        updateResultsCount(0);
        return;
    }

    emptyState.style.display = 'none';
    suppliersList.style.display = 'flex';
    suppliersList.innerHTML = '';

    // Découpage de la page courante. La recherche remet à la première page,
    // sinon on resterait sur une page qui n'existe plus dans le nouveau filtre.
    const pages = Math.max(1, Math.ceil(suppliersToShow.length / SUPPLIERS_PER_PAGE));
    if (suppliersPage > pages) suppliersPage = 1;
    const start = (suppliersPage - 1) * SUPPLIERS_PER_PAGE;
    const pageItems = suppliersToShow.slice(start, start + SUPPLIERS_PER_PAGE);

    if (typeof renderPager === 'function') {
        renderPager('suppliersPager', {
            page: suppliersPage,
            pages,
            total: suppliersToShow.length,
            onPage: (p) => { suppliersPage = p; displaySuppliers(suppliersToShow); }
        });
    }

    pageItems.forEach(supplier => {
        const supplierCard = createSupplierCard(supplier);
        suppliersList.appendChild(supplierCard);
    });

    updateResultsCount(suppliersToShow.length);
}

// Créer une carte fournisseur
function createSupplierCard(supplier) {
    const col = document.createElement('div');
    col.className = 'col-lg-6 col-xl-4 mb-4';

    const sid = supplier.supplier_id || supplier.id;
    col.innerHTML = `
        <div class="card h-100 shadow-sm border-0 supplier-card">
            <div class="card-body">
                <div class="d-flex justify-content-between align-items-start mb-3">
                    <div class="flex-grow-1">
                        <h5 class="card-title mb-1 text-primary">
                            <i class="bi bi-building me-2"></i>
                            <a href="/supplier/${sid}" class="text-decoration-none text-primary">
                                ${escapeHtml(supplier.name)}
                            </a>
                        </h5>
                        <small class="text-muted">ID: ${sid}</small>
                    </div>
                    <div class="dropdown">
                        <button class="btn btn-outline-secondary btn-sm dropdown-toggle" type="button" data-bs-toggle="dropdown">
                            <i class="bi bi-three-dots"></i>
                        </button>
                        <ul class="dropdown-menu">
                            <li>
                                <a class="dropdown-item" href="/supplier/${sid}">
                                    <i class="bi bi-eye me-2"></i>
                                    Voir détails
                                </a>
                            </li>
                            <li>
                                <button class="dropdown-item" onclick="editSupplier(${sid})">
                                    <i class="bi bi-pencil me-2"></i>
                                    Modifier
                                </button>
                            </li>
                            <li><hr class="dropdown-divider" /></li>
                            <li>
                                <button class="dropdown-item text-danger" onclick="deleteSupplier(${sid})">
                                    <i class="bi bi-trash me-2"></i>
                                    Supprimer
                                </button>
                            </li>
                        </ul>
                    </div>
                </div>
                
                <div class="supplier-info">
                    ${supplier.contact_person ? `
                        <div class="mb-2">
                            <i class="bi bi-person text-muted me-2"></i>
                            <span class="text-muted small">Contact:</span>
                            <div class="ms-4">${escapeHtml(supplier.contact_person)}</div>
                        </div>
                    ` : ''}
                    
                    ${supplier.email ? `
                        <div class="mb-2">
                            <i class="bi bi-envelope text-muted me-2"></i>
                            <span class="text-muted small">Email:</span>
                            <div class="ms-4">
                                <a href="mailto:${supplier.email}" class="text-decoration-none">
                                    ${escapeHtml(supplier.email)}
                                </a>
                            </div>
                        </div>
                    ` : ''}
                    
                    ${supplier.phone ? `
                        <div class="mb-2">
                            <i class="bi bi-telephone text-muted me-2"></i>
                            <span class="text-muted small">Téléphone:</span>
                            <div class="ms-4">
                                <a href="tel:${supplier.phone}" class="text-decoration-none">
                                    ${escapeHtml(supplier.phone)}
                                </a>
                            </div>
                        </div>
                    ` : ''}
                    
                    ${supplier.address ? `
                        <div class="mb-2">
                            <i class="bi bi-geo-alt text-muted me-2"></i>
                            <span class="text-muted small">Adresse:</span>
                            <div class="ms-4 text-break">${escapeHtml(supplier.address)}</div>
                        </div>
                    ` : ''}
                </div>
            </div>
            
            <!--
              Pas de pied d'actions : le menu « … » de l'en-tête porte déjà
              « Voir détails », « Modifier » et « Supprimer ». Les deux jeux
              coexistaient, et le bouton de modification y prenait la pastille
              claire réservée à l'action principale de la page.
            -->
        </div>
    `;

    return col;
}

// Filtrer les fournisseurs
function filterSuppliers() {
    const searchTerm = document.getElementById('searchInput').value.toLowerCase();

    if (!searchTerm) {
        displaySuppliers(suppliers);
        return;
    }

    const filtered = suppliers.filter(supplier =>
        supplier.name.toLowerCase().includes(searchTerm) ||
        (supplier.contact_person && supplier.contact_person.toLowerCase().includes(searchTerm)) ||
        (supplier.email && supplier.email.toLowerCase().includes(searchTerm)) ||
        (supplier.phone && supplier.phone.includes(searchTerm)) ||
        (supplier.address && supplier.address.toLowerCase().includes(searchTerm))
    );

    displaySuppliers(filtered);
}

// Mettre à jour les statistiques
function updateStatistics() {
    const totalSuppliers = suppliers.length;
    const activeSuppliers = suppliers.filter(s => s.status === 'active' || !s.status).length;
    const suppliersWithEmail = suppliers.filter(s => s.email).length;
    const suppliersWithPhone = suppliers.filter(s => s.phone).length;

    // Mettre à jour les éléments avec vérification d'existence
    const totalElement = document.getElementById('totalSuppliers');
    const activeElement = document.getElementById('activeSuppliers');
    const emailElement = document.getElementById('suppliersWithEmail');
    const phoneElement = document.getElementById('suppliersWithPhone');

    if (totalElement) totalElement.textContent = totalSuppliers;
    if (activeElement) activeElement.textContent = activeSuppliers;
    if (emailElement) emailElement.textContent = suppliersWithEmail;
    if (phoneElement) phoneElement.textContent = suppliersWithPhone;
}

// Mettre à jour le compteur de résultats
function updateResultsCount(count) {
    const resultsCount = document.getElementById('resultsCount');
    if (resultsCount) {
        resultsCount.textContent = `${count} fournisseur${count !== 1 ? 's' : ''} trouvé${count !== 1 ? 's' : ''}`;
    }
}

// Basculer l'affichage du formulaire
function toggleForm() {
    const form = document.getElementById('supplierForm');
    const btn = document.getElementById('toggleFormBtn');

    if (form.style.display === 'none') {
        form.style.display = 'block';
        btn.innerHTML = '<i class="bi bi-x-circle me-2"></i>Annuler';
        btn.className = 'btn btn-outline-secondary btn-lg shadow-sm';
    } else {
        resetForm();
    }
}

// Réinitialiser le formulaire
function resetForm() {
    const form = document.getElementById('supplierFormElement');
    const formContainer = document.getElementById('supplierForm');
    const btn = document.getElementById('toggleFormBtn');
    const formTitle = document.getElementById('formTitle');
    const submitBtnText = document.getElementById('submitBtnText');

    form.reset();
    formContainer.style.display = 'none';
    btn.innerHTML = '<i class="bi bi-plus-circle me-2"></i>Ajouter un Fournisseur';
    btn.className = 'btn btn-success btn-lg shadow-sm';

    formTitle.innerHTML = '<i class="bi bi-plus-circle me-2"></i>Ajouter un Fournisseur';
    submitBtnText.textContent = 'Ajouter';

    isEditMode = false;
    currentSupplierId = null;
    document.getElementById('supplierId').value = '';
}

// Modifier un fournisseur
function editSupplier(id) {
    const supplier = suppliers.find(s => (s.supplier_id || s.id) === id);
    if (!supplier) return;

    // Remplir le formulaire
    document.getElementById('supplierId').value = id;
    document.getElementById('name').value = supplier.name || '';
    document.getElementById('contact_person').value = supplier.contact_person || '';
    document.getElementById('email').value = supplier.email || '';
    document.getElementById('phone').value = supplier.phone || '';
    document.getElementById('address').value = supplier.address || '';

    // Mettre à jour l'interface
    const formContainer = document.getElementById('supplierForm');
    const btn = document.getElementById('toggleFormBtn');
    const formTitle = document.getElementById('formTitle');
    const submitBtnText = document.getElementById('submitBtnText');

    formContainer.style.display = 'block';
    btn.innerHTML = '<i class="bi bi-x-circle me-2"></i>Annuler';
    btn.className = 'btn btn-outline-secondary btn-lg shadow-sm';

    formTitle.innerHTML = '<i class="bi bi-pencil me-2"></i>Modifier le Fournisseur';
    submitBtnText.textContent = 'Mettre à jour';

    isEditMode = true;
    currentSupplierId = id;

    // Scroll vers le formulaire
    formContainer.scrollIntoView({ behavior: 'smooth' });
}

// Gérer la soumission du formulaire
async function handleFormSubmit(e) {
    e.preventDefault();

    const supplierData = {
        name: document.getElementById('name').value,
        contact_person: document.getElementById('contact_person').value,
        email: document.getElementById('email').value,
        phone: document.getElementById('phone').value,
        address: document.getElementById('address').value
    };

    // Validation
    if (!supplierData.name.trim()) {
        showError('Le nom du fournisseur est obligatoire');
        return;
    }

    try {
        let response;

        if (isEditMode && currentSupplierId) {
            response = await axios.put(`/api/suppliers/${currentSupplierId}/`, supplierData);
        } else {
            response = await axios.post('/api/suppliers/', supplierData);
        }

        if (response.status && response.status >= 400) {
            throw new Error('Erreur lors de l\'enregistrement du fournisseur');
        }

        showSuccess(isEditMode ? 'Fournisseur modifié avec succès' : 'Fournisseur ajouté avec succès');
        resetForm();
        loadSuppliers();
    } catch (error) {
        console.error('Erreur:', error);
        showError(error.response?.data?.detail || 'Erreur lors de l\'enregistrement du fournisseur');
    }
}

// Supprimer un fournisseur
function deleteSupplier(id) {
    currentSupplierId = id;
    const modal = new bootstrap.Modal(document.getElementById('deleteModal'));
    modal.show();
}

// Confirmer la suppression
async function confirmDelete() {
    if (!currentSupplierId) return;

    try {
        await axios.delete(`/api/suppliers/${currentSupplierId}/`);

        showSuccess('Fournisseur supprimé avec succès');

        // Fermer la modal
        const modal = bootstrap.Modal.getInstance(document.getElementById('deleteModal'));
        modal.hide();

        // Recharger les données
        loadSuppliers();
        currentSupplierId = null;
    } catch (error) {
        console.error('Erreur:', error);
        showError(error.response?.data?.detail || 'Erreur lors de la suppression du fournisseur');
    }
}

// Afficher le loading
function showLoading() {
    // Ne pas afficher d'indicateur de chargement pour une expérience instantanée
}

// Masquer le loading
function hideLoading() {
    // Récupérer le conteneur
    const container = document.getElementById('suppliersContainer');

    // Vérifier si le conteneur contient l'indicateur de chargement
    const loadingSpinner = container.querySelector('.spinner-border');

    if (loadingSpinner) {
        // Vider le conteneur de chargement
        container.innerHTML = '';

        // Recréer l'état vide dans le conteneur
        const emptyState = document.createElement('div');
        emptyState.id = 'emptyState';
        emptyState.className = 'text-center py-5';
        emptyState.innerHTML = `
            <i class="bi bi-truck display-1 text-muted"></i>
            <h4 class="text-muted mt-3">Aucun fournisseur</h4>
            <p class="text-muted">Commencez par ajouter votre premier fournisseur</p>
        `;
        container.appendChild(emptyState);
    }

    // S'assurer que suppliersList existe en dehors du conteneur
    let suppliersList = document.getElementById('suppliersList');
    if (!suppliersList) {
        // Créer la liste des fournisseurs après le conteneur
        const parentElement = container.parentElement;
        suppliersList = document.createElement('div');
        suppliersList.id = 'suppliersList';
        suppliersList.className = 'row';
        suppliersList.style.display = 'none';
        parentElement.insertBefore(suppliersList, container.nextSibling);
    }
}

/*
  Ce que cet écran recharge après une écriture.

  Déclaré ici et appelé par la couche HTTP (voir static/js/http.js) : toute
  création, modification ou suppression réussie relance ce chargement, si bien
  que la liste ne montre plus une valeur périmée pendant que la modale, elle,
  est à jour.
*/
window.rafraichirDonnees = function () {
    const taches = [];
    if (typeof loadSuppliers === 'function') taches.push(loadSuppliers());
    return Promise.allSettled(taches);
};

File: app/routers/suppliers.py
from fastapi import APIRouter, HTTPException, Depends, Query, Response
from sqlalchemy.orm import Session
from sqlalchemy import or_, desc, func
from typing import Optional, List
from decimal import Decimal

from ..database import get_db, Supplier
from ..schemas import SupplierQuickCreate, SupplierResponse
from ..auth import get_current_user, User

router = APIRouter(
    prefix="/api/suppliers",
    tags=["suppliers"],
    dependencies=[Depends(get_current_user)]
)

@router.get("/", response_model=List[SupplierResponse])
async def get_suppliers(
    skip: int = Query(0, ge=0, description="Nombre d'éléments à ignorer"),
    limit: int = Query(100, ge=1, le=1000, description="Nombre d'éléments à récupérer"),
    search: Optional[str] = Query(None, description="Recherche par nom, téléphone ou email"),
    response: Response = None,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Récupérer la liste des fournisseurs avec pagination et recherche"""
    
    try:
        query = db.query(Supplier)
        
        # Appliquer les filtres de recherche
        if search and search.strip():
            search_term = f"%{search.strip().lower()}%"
            query = query.filter(
                or_(
                    func.lower(Supplier.name).like(search_term),
                    func.lower(Supplier.phone).like(search_term),
                    func.lower(Supplier.email).like(search_term)
                )
            )
        
        # Le total est publié en en-tête plutôt que dans le corps : la réponse
        # est typée `List[SupplierResponse]`, l'envelopper casserait tous les
        # appelants existants. La pagination de l'écran s'en sert pour savoir
        # combien de pages afficher.
        total = query.count()

        # Trier par nom
        query = query.order_by(Supplier.name)

        # Appliquer pagination
        suppliers = query.offset(skip).limit(limit).all()

        response.headers["X-Total-Count"] = str(total)
        return suppliers
        
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Erreur lors de la récupération: {str(e)}")

@router.get("/{supplier_id}", response_model=SupplierResponse)
async def get_supplier(
    supplier_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Récupérer un fournisseur spécifique"""
    
    supplier = db.query(Supplier).filter(Supplier.supplier_id == supplier_id).first()
    if not supplier:
        raise HTTPException(status_code=404, detail="Fournisseur introuvable")
    
    return supplier

@router.post("/", response_model=SupplierResponse)
async def create_supplier(
    supplier_data: SupplierQuickCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Créer un nouveau fournisseur"""
    
    try:
        # Vérifier si un fournisseur avec ce nom existe déjà
        existing = db.query(Supplier).filter(
            func.lower(Supplier.name) == func.lower(supplier_data.name)
        ).first()
        
        if existing:
            raise HTTPException(
                status_code=400, 
                detail=f"Un fournisseur avec le nom '{supplier_data.name}' existe déjà"
            )
        
        # Créer le nouveau fournisseur
        db_supplier = Supplier(
            name=supplier_data.name,
            phone=supplier_data.phone,
            email=supplier_data.email,
            address=supplier_data.address
        )
        
        db.add(db_supplier)
        db.commit()
        db.refresh(db_supplier)
        
        return db_supplier
        
    except HTTPException:
        raise
    except Exception as e:
        db.rollback()
        raise HTTPException(status_code=500, detail=f"Erreur lors de la création: {str(e)}")

@router.put("/{supplier_id}", response_model=SupplierResponse)
async def update_supplier(
    supplier_id: int,
    supplier_data: SupplierQuickCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Mettre à jour un fournisseur"""
    
    try:
        # Récupérer le fournisseur existant
        supplier = db.query(Supplier).filter(Supplier.supplier_id == supplier_id).first()
        if not supplier:
            raise HTTPException(status_code=404, detail="Fournisseur introuvable")
        
        # Vérifier les doublons de nom (sauf pour le fournisseur actuel)
        existing = db.query(Supplier).filter(
            Supplier.supplier_id != supplier_id,
            func.lower(Supplier.name) == func.lower(supplier_data.name)
        ).first()
        
        if existing:
            raise HTTPException(
                status_code=400,
                detail=f"Un autre fournisseur avec le nom '{supplier_data.name}' existe déjà"
            )
        
        # Mettre à jour les champs
        supplier.name = supplier_data.name
        supplier.phone = supplier_data.phone
        supplier.email = supplier_data.email
        supplier.address = supplier_data.address
        
        db.commit()
        db.refresh(supplier)
        
        return supplier
        
    except HTTPException:
        raise
    except Exception as e:
        db.rollback()
        raise HTTPException(status_code=500, detail=f"Erreur lors de la mise à jour: {str(e)}")

@router.delete("/{supplier_id}")
async def delete_supplier(
    supplier_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Supprimer un fournisseur"""
    
    try:
        # Récupérer le fournisseur
        supplier = db.query(Supplier).filter(Supplier.supplier_id == supplier_id).first()
        if not supplier:
            raise HTTPException(status_code=404, detail="Fournisseur introuvable")
        
        # Vérifier s'il y a des factures liées
        from ..database import SupplierInvoice
        invoice_count = db.query(SupplierInvoice).filter(
            SupplierInvoice.supplier_id == supplier_id
        ).count()
        
        if invoice_count > 0:
            raise HTTPException(
                status_code=400,
                detail=f"Impossible de supprimer: {invoice_count} facture(s) sont liées à ce fournisseur"
            )
        
        # Supprimer le fournisseur
        db.delete(supplier)
        db.commit()
        
        return {"message": "Fournisseur supprimé avec succès"}
        
    except HTTPException:
        raise
    except Exception as e:
        db.rollback()
        raise HTTPException(status_code=500, detail=f"Erreur lors de la suppression: {str(e)}")

@router.get("/search/suggestions")
async def get_supplier_suggestions(
    q: str = Query(..., min_length=2, description="Terme de recherche"),
    limit: int = Query(10, ge=1, le=50, description="Nombre de suggestions"),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """Récupérer des suggestions de fournisseurs pour l'autocomplétion"""
    
    try:
        search_term = f"%{q.strip().lower()}%"
        
        suppliers = db.query(Supplier).filter(
            or_(
                func.lower(Supplier.name).like(search_term),
                func.lower(Supplier.phone).like(search_term),
                func.lower(Supplier.email).like(search_term)
            )
        ).order_by(Supplier.name).limit(limit).all()
        
        return [
            {
                "id": supplier.supplier_id,
                "name": supplier.name,
                "phone": supplier.phone,
                "email": supplier.email
            }
            for supplier in suppliers
        ]
        
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Erreur lors de la recherche: {str(e)}")
